MathLib/LinAlg/Eigen/EigenTools.h

```diff
@@ -11,12 +11,14 @@
 #define EIGENTOOLS_H_
 
 #include <vector>
+#include <Eigen/Sparse> //for RawMatrixType::Index
 
 namespace MathLib
 {
 class EigenMatrix;
 class EigenVector;
 
+using RawMatrixType = Eigen::SparseMatrix<double, Eigen::RowMajor>;
```

Comment:
Just want to take the advantage of the forward class declaration, and to avoid include the header of the matrix classes that have the definition of IndexType.